ros2_batch_job/__main__.py

```diff
@@ -273,7 +283,56 @@ def get_args(sysargv=None):
     return args
 
 
-def process_coverage(args, job):
+def include_python_coverage_in_report(args, package=None):
+    # no arguments means report all packages in buildspace
+    root_directory_to_report = args.buildspace
+    if package:
+        root_directory_to_report = os.path.join(root_directory_to_report, package)
+
+    coverage_xml_files = glob.glob(
+        os.path.join(root_directory_to_report, '**', 'coverage.xml'),
+        recursive=True)
+    for coverage_xml_path in coverage_xml_files:
+        # python coverage detected: move the coverage.xml file to buildspace to be reported
+        dest_file_to_report = os.path.join(args.buildspace, coverage_xml_path.replace(os.path.sep, '.'))
+        copyfile(coverage_xml_path, dest_file_to_report)
+
+
+def filter_unit_coverage(args, coverage_info_file, packages_to_filter):
+    build_paths_collection = []
+    src_paths_collection = []
+    for package_name in packages_to_filter:
+        # check if it is a python package generating its own coverage.xml
+        include_python_coverage_in_report(args, package_name)
+        # collect paths to run lcov in order to process C/C++ coverage information
+        cmd = [
+            args.colcon_script,
+            'list',
+            '--paths-only',
+            '--base-paths', args.sourcespace,
+            '--packages-select', package_name]
+        print(cmd)
+        src_path = subprocess.check_output(cmd).decode('ascii').strip()
```

Comment:
Depending on the number of packages it looks inefficient to invoke a subprocess for each. Why not invoke one command with all desired package names?

Reply:
Run `colcon list` just once ccb0ee8